src/lib/y2storage/filesystems/legacy_nfs.rb

```diff
@@ -0,0 +1,230 @@
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/filesystems/nfs"
+
+module Y2Storage
+  module Filesystems
+    # Class to represent an NFS mount, as defined in the old y2-storage.
+    # This class is useful to connect code that uses objects of the new {Nfs}
+    # class and code using the old hash-based format (the so-called TargetMap).
+    # In particular to connect y2-storage-ng and y2-nfs-client.
+    class LegacyNfs
+      include Yast::Logger
+
+      # Devicegraph to use by default in the subsequent operations
+      # @return [Devicegraph]
+      attr_accessor :default_devicegraph
+
+      # Server name or IP address of the NFS share
+      # @return [String]
+      attr_reader :server
+
+      # Remote path of the NFS share
+      # @return [String]
+      attr_reader :path
+
+      # Previous server name or IP, used when the remote share is changed
+      # (which strictly speaking means replacing the NFS mount with a new one)
+      #
+      # @see #server
+      #
+      # @return [String]
+      attr_reader :old_server
+
+      # Previous share path
+      #
+      # @see #old_server
+      # @see #path
+      #
+      # @return [String]
+      attr_reader :old_path
+
+      # Local mount point path
+      # @return [String]
+      attr_reader :mountpoint
+
+      # Options field for fstab
+      # @return [String]
+      attr_reader :fstopt
+
+      # Creates a new object from a hash with the legacy fields used in
+      # TargetMap-based code.
+      #
+      # @return [LegacyNfs]
+      def self.new_from_hash(legacy_hash)
+        legacy = new
+        legacy.initialize_from_hash(legacy_hash)
+        legacy
+      end
+
+      # Creates a new object with an {Nfs} object as starting point.
+      #
+      # @return [LegacyNfs]
+      def self.new_from_nfs(nfs)
+        legacy = new
+        legacy.initialize_from_nfs(nfs)
+        legacy
+      end
+
+      # Hash representation of the object, with the fields used in
+      # TargetMap-based code (like y2-nfs-client).
+      #
+      # @return [Hash]
+      def to_hash
+        hash = {
+          "device"  => share_string(server, path),
+          "mount"   => mountpoint,
+          "fstopt"  => fstopt,
+          # TODO: libstorage-ng does not distinguish different NFS versions, so
+          # for the time being, always use :nfs here
+          "used_fs" => :nfs
+        }
+        hash["old_device"] = share_string(old_server, old_path) if share_changed?
+        hash
+      end
+
+      # Creates an {Nfs} object, equivalent to this one, in the devicegraph
+      #
+      # @raise [ArgumentError] if no devicegraph is given and no default
+      #   devicegraph has been previously defined
+      #
+      # @param devicegraph [Devicegraph, nil] if nil, the default devicegraph
+      #   will be used
+      # @return [Nfs] the new device
+      def create_nfs_device(devicegraph = nil)
+        graph = check_devicegraph(devicegraph)
```

Comment:
np: I would replace `check` with `find`. At a first sight, I though that you were doing some "health check" to the devicegraph :)

Reply:
Because it actually checks the validity of the `devicegraph` argument (and raises an exception if none provided). So I would rather change it to `check_devicegraph_arg` or `check_devicegraph_param` than to `find_devicegraph`

Reply:
Ok, I am fine with any solution, but not `check_devicegraph` :)